tests/test_kernel_language.py

```diff
@@ -365,6 +366,51 @@ class DensParticle(ptype[mode]):
     assert np.allclose(pset[0].density, 1022.85377)
 
 
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_EOSseawaterproperties_kernels(mode):
+
+    def generate_fieldset(xdim=2, ydim=2, zdim=2, tdim=1):
+        lon = np.linspace(0., 10., xdim, dtype=np.float32)
+        lat = np.linspace(0., 10., ydim, dtype=np.float32)
+        depth = np.linspace(0, 2000, zdim, dtype=np.float32)
+        time = np.zeros(tdim, dtype=np.float64)
+        U = np.ones((tdim, zdim, ydim, xdim))
+        V = np.ones((tdim, zdim, ydim, xdim))
+        abs_salinity = 40 * np.ones((tdim, zdim, ydim, xdim))
+        temperature = 40 * np.ones((tdim, zdim, ydim, xdim))
+        potemperature = 36.89073 * np.ones((tdim, zdim, ydim, xdim))
+        dimensions = {'lat': lat, 'lon': lon, 'depth': depth, 'time': time}
+        data = {'U': np.array(U, dtype=np.float32), 'V': np.array(V, dtype=np.float32),
+                'psu_salinity': np.array(abs_salinity, dtype=np.float32),
+                'temperature': np.array(temperature, dtype=np.float32),
+                'potemperature': np.array(potemperature, dtype=np.float32)}
+        return (data, dimensions)
+
+    data, dimensions = generate_fieldset()
+    fieldset = FieldSet.from_data(data, dimensions)
+    fieldset.add_constant('refpressure', np.float(0))
+
+    class PoTempParticle(ptype[mode]):
+        potemp = Variable('potemp', dtype=np.float32)
+        pressure = Variable('pressure', dtype=np.float32, initial=10000)
+    pset = ParticleSet(fieldset, pclass=PoTempParticle, lon=5, lat=5, depth=1000)
+    pset.execute(ptemp_from_temp, runtime=0, dt=0)
+    assert np.allclose(pset[0].potemp, 36.89073)
+
+    class TempParticle(ptype[mode]):
+        temp = Variable('temp', dtype=np.float32)
+        pressure = Variable('pressure', dtype=np.float32, initial=10000)
+    pset = ParticleSet(fieldset, pclass=TempParticle, lon=5, lat=5, depth=1000)
+    pset.execute(temp_from_ptemp, runtime=0, dt=0)
+    assert np.allclose(pset[0].temp, 40)
+
+    class TPressureParticle(ptype[mode]):
+        pressure = Variable('pressure', dtype=np.float32)
+    pset = ParticleSet(fieldset, pclass=TempParticle, lon=5, lat=30, depth=7321.45)
+    pset.execute(pressure_from_latdepth, runtime=0, dt=0)
+    assert np.allclose(pset[0].pressure, 7500.0065130118019)
```

Comment:
Not sure why the large number of significant digits here. Should this be 7500? If so, perhaps this code helps
```suggestion
    assert np.allclose(pset[0].pressure, 7500, atol=1e-2)
```